axon-framework/monitoring/tracing.md

```diff
@@ -0,0 +1,535 @@
+
+### Snapshotting
+
+Snapshotting is done in a separate root trace, due to the fact that it's an asynchronous action and has no user impact.
+However, it can still be useful to measure the performance of snapshotting and see when it is triggered.
+The root trace of the `Snapshotter` invocation will be linked to the command handling span 
+after which the snapshot was scheduled to be created.
+
+| Trace name                                                                  | Description                                                                                                           |
+|-----------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------|
+| `${SnapshotterClass}.createSnapshot($aggregateClass)`                       | A snapshot creation task is being submitted. Depending on performance, the executor might take a while to pick it up. |
+| `${SnapshotterClass}.createSnapshot($aggregateClass, $aggregateIdentifier)` | The `Snapshotter` is now creating the snapshot.                                                                       |
+
+Note that the root trace does not contain the aggregate identifier,
+so any snapshot invocation of the same aggregate type is grouped together by the APM tool.
+
+### Sagas
+Sagas are a special type of event processor that can invoke multiple saga's for a single event. 
+Because of this the `AbstractSagaManager` has been instructed with additional tracing information.
+These spans are descendants of an event processor span that invokes the manager.
+
+| Trace name                                                                  | Description                                          |
+|-----------------------------------------------------------------------------|------------------------------------------------------|
+| `SagaManager[${SagaTypeName}].invokeSaga ${sagaIdentifier}`                 | A matching saga has been found and is being invoked. |
+| `SagaManager[${SagaTypeName}].startNewSaga`                                 | The manager is constructing a new saga.              |
+
+
+### Queries
+Queries support tracing in all of their forms. In order to be clear about how they work, 
+this section is split based upon the query's type. 
+For all types, the created spans will differ based on whether Axon Server is used or not. 
+The spans that are only available with Axon Server are marked as such.
+
+#### Direct queries
+Direct queries fetch a single result (either a single item or a single list) and receive no updates. 
+Their traces will look like the following table:
+
+| Trace name                                                | Description                                                                           |
```

Comment:
I'd have a similar code-block suggestion for the AxonServer and non-AxonServer samples, just as with the Commands section.